### PR TITLE
[API] Optimizing check_size.rs

### PR DESCRIPTION
**Pull request by thouravi**
- Instead of using a separate `PostSizeLimit` struct and `PostSizeLimitEndpoint` struct, you can combine them into a single struct that implements the `Middleware` and `Endpoint` traits. This will reduce the amount of code and make the implementation more concise.

- Instead of comparing the request method against `Method::POST` using the != operator, you can use the `Method::is_post` method, which is more efficient and readable.

- Instead of using the `ok_or` method to handle the case when the `ContentLength` header is missing, you can use the `try_get` method and check for `None` directly. This will avoid creating an instance of `SizedLimitError::MissingContentLength` unnecessarily.

### CODE REVIEW

File: api/src/check_size.rs [hunk 1]
```diff
@@ -11,45 +11,37 @@ use poem::{
...
         }
 
-        let content_length = req
-            .headers()
-            .typed_get::<headers::ContentLength>()
-            .ok_or(SizedLimitError::MissingContentLength)?;
+        let content_length = match req.headers().typed_try_get::<headers::ContentLength>() {
+            Some(content_length) => content_length,
+            None => return Err(SizedLimitError::MissingContentLength.into()),
+        };
```
Comment (gregnazario) [inline api/src/check_size.rs:44]: Readability is not quite as good here.  Wondering if you did any benchmarking to see what the optimizations are here?

File: api/src/check_size.rs [hunk 2]
```diff
@@ -11,45 +11,37 @@ use poem::{
 /// This middleware confirms that the Content-Length header is set and the
 /// value is within the acceptable range. It only applies to POST requests.
 pub struct PostSizeLimit {
+    inner: E,
     max_size: u64,
 }
```
Comment (gregnazario) [inline api/src/check_size.rs:16]: I think you need a generic here?  Otherwise, E isn't defined

**Review by gregnazario — COMMENTED** [review]
Hey thanks for these updates!  I've kicked off the test build.

File: api/src/check_size.rs (hunk 1 above)
Comment (thouravi) [reply api/src/check_size.rs:44]: Haven't done any benchmarking yet, will work on it. 
About the readability part, I agree with that.

**Comment by davidiw** [comment]
Please test PRs locally before submitting them for review. This code doesn't compile, we cannot be your compilers.

**Pull request closed without merging**